nano/nano_node/daemon.hpp

```diff
@@ -2,13 +2,27 @@
 
 namespace nano
 {
+class node;
+class rpc;
 class node_flags;
+namespace ipc
+{
+	class ipc_server;
+}
 }
 namespace nano_daemon
 {
 class daemon
 {
 public:
-	void run (boost::filesystem::path const &, nano::node_flags const & flags);
+	daemon (boost::filesystem::path const & data_path, nano::node_flags const & flags);
+	~daemon ();
```

Comment:
As you are not using a virtual destructor (which I think is correct), the class should be `final` so it cannot be subclassed.